src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6181,6 +6204,68 @@ IOperation popTarget(IOperation originalTarget)
                         throw ExceptionUtilities.UnexpectedValue(originalTarget.Kind);
                 }
             }
+
+            static bool onlyContainsEmptyLeafNestedInitializers(IMemberInitializerOperation memberInitializer)
+            {
+                return memberInitializer.Initializer is IObjectOrCollectionInitializerOperation initializer
```

Comment:
>IObjectOrCollectionInitializerOperation [](http://example.com/codeflow?start=56&length=39)

It looks like lowering doesn't optimize collection initializers this way.

Reply:
The IOperation representation differs from the initial bound nodes. There's only one kind of IOperation node to represent object vs. collection initializers.